repair_shards.c

```diff
@@ -137,10 +137,7 @@ master_copy_shard_placement(PG_FUNCTION_ARGS)
 	}
 
 	/* the placement is repaired, so return to finalized state */
-	DeleteShardPlacementRow(targetPlacement->id);
```

Comment:
Is there a reason to diverge from this pattern? I realize we now need to do so because of the auto-incrementing sequence. Are there any advantages/disadvantages vs the delete/insert vs update pattern? Maybe we loop in @ozgune on this?